sdk/translation/azure-ai-translation-text/src/main/java/com/azure/ai/translation/text/TranslatorRegionAuthenticationPolicy.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.translation.text;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import java.util.Objects;
+import reactor.core.publisher.Mono;
+
+/**
+ * Adds MT custom authentication headers to the requests.
+ */
+public class TranslatorRegionAuthenticationPolicy implements HttpPipelinePolicy {
```

Comment:
Make this class non-public, the constructor is package-private so no one except the SDK code will be able to create this.

Reply:
I have tried it and got linting error:

```bash
TranslatorRegionAuthenticationPolicy.java",19,1,"Class 'TranslatorRegionAuthenticationPolicy' implementing 'HttpPipelinePolicy' and should be a public class","com.azure.tools.checkstyle.checks.HttpPipelinePolicyCheck"
```

Shall I just disable this rule for the given file?

Reply:
Oh yeah, just add a new Checkstyle exception into `/eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml`. Here is a link to an example of this:

https://github.com/Azure/azure-sdk-for-java/blob/main/eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml#L336